charts/dvpe-deployment-gloo/templates/secret.yaml

```diff
@@ -33,6 +33,26 @@ spec:
 
 {{- end }}
 
+{{- if and .Values.externalSecrets.oidc.key .Values.gloo.authConfig.spec.configs.clientCredentialsPlugin.config.clientId }}
+---
+apiVersion: 'kubernetes-client.io/v1'
+kind: ExternalSecret
+metadata:
+  name: {{ $serviceName }}-oidc-m2m-secrets
+  namespace: {{ .Release.Namespace }}
+spec:
+  backendType: secretsManager
+  template:
+    metadata:
+      annotations:
+        resource_kind: '*v1.Secret'
```

Comment:
`resource_kind` can be removed. Will be added by Kubernetes. If we don't have any further `annotations` we can remove the whole `template` section

Reply:
I was told that it is actually crucial to add that explicitly - it necessary for external config. Am I right @jensbac?

Reply:
This `resource_kind` annotation is essential to get Gloo using the `Secret` derived from the generated `ExternalSecret`. When getting the `ExternalSecrets` approach working for Gloo ExtAuth, this annotation was the key point.
I found this out by experiments and comparison between manually created working secrets (`webeam_oidc` present before) and the `ExternalSecrets` generated ones.

Reply:
ok good to know :) Than we should keep it